server/websocket.go

```diff
@@ -419,36 +421,49 @@ func (p *Plugin) handleJoin(userID, connID, channelID, title string) error {
 			},
 		}
 		if err := p.rtcdManager.Send(msg, channelID); err != nil {
-			p.LogError(fmt.Errorf("failed to send client message: %w", err).Error())
+			return fmt.Errorf("failed to send client join message: %w", err)
 		}
 	} else {
 		if handlerID == p.nodeID {
-			wg.Add(1)
-			go func() {
-				defer wg.Done()
```

Comment:
This was pure technical debt at this point since we made the initialization non-blocking a while ago so no need for a goroutine anymore. We can now easily return errors to the caller.